internal/serve/httphandler/receiver_send_otp_handler.go

```diff
@@ -97,77 +112,136 @@ func (h ReceiverSendOTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
+	}
+
+	sendOTPMessageTpl, err := template.New("").Parse(otpMessageTemplate)
 	if err != nil {
-		httperror.InternalError(ctx, "Cannot update OTP for receiver wallet", err, nil).Render(w)
-		return
+		return fmt.Errorf("cannot parse OTP template: %w", err)
 	}
 
-	if numberOfUpdatedRows < 1 {
-		log.Ctx(ctx).Warnf("updated no rows in ReceiverSendOTPHandler, please verify if the provided phone number (%s) and client_domain (%s) are both valid", truncatedPhoneNumber, sep24Claims.ClientDomainClaim)
-	} else {
-		sendOTPData := ReceiverSendOTPData{
-			OTP:              newOTP,
-			OrganizationName: organization.Name,
-		}
-
-		otpMessageTemplate := organization.OTPMessageTemplate + OTPMessageDisclaimer
-		if !strings.Contains(organization.OTPMessageTemplate, "{{.OTP}}") {
-			// Adding the OTP code to the template
-			otpMessageTemplate = fmt.Sprintf(`{{.OTP}} %s`, strings.TrimSpace(otpMessageTemplate))
-		}
-
-		sendOTPMessageTpl, err := template.New("").Parse(otpMessageTemplate)
-		if err != nil {
-			httperror.InternalError(ctx, "Cannot parse OTP template", err, nil).Render(w)
-			return
-		}
-
-		builder := new(strings.Builder)
-		if err = sendOTPMessageTpl.Execute(builder, sendOTPData); err != nil {
-			httperror.InternalError(ctx, "Cannot execute OTP template", err, nil).Render(w)
-			return
-		}
-
-		msg := message.Message{
-			ToPhoneNumber: receiverSendOTPRequest.PhoneNumber,
-			Message:       builder.String(),
-		}
-
-		// TODO: SDP-1296 - support multiple channels for OTP
-		log.Ctx(ctx).Infof("sending OTP message to phone number: %s", truncatedPhoneNumber)
-		err = h.MessageDispatcher.SendMessage(ctx, msg, organization.MessageChannelPriority)
-		if err != nil {
-			httperror.InternalError(ctx, "Cannot send OTP message", err, nil).Render(w)
-			return
-		}
-	}
-
-	response := ReceiverSendOTPResponseBody{
-		Message:           "if your phone number is registered, you'll receive an OTP",
-		VerificationField: verificationField,
+	sendOTPData := ReceiverSendOTPData{
+		OTP:              otp,
+		OrganizationName: organization.Name,
 	}
-	httpjson.RenderStatus(w, http.StatusOK, response, httpjson.JSON)
+
+	builder := new(strings.Builder)
+	if err = sendOTPMessageTpl.Execute(builder, sendOTPData); err != nil {
+		return fmt.Errorf("cannot execute OTP template: %w", err)
+	}
+
+	msg := message.Message{Message: builder.String()}
+	switch contactType {
+	case data.ReceiverContactTypeSMS:
+		msg.ToPhoneNumber = contactInfo
+	case data.ReceiverContactTypeEmail:
+		msg.ToEmail = contactInfo
+		msg.Title = "Your One-Time Password: " + otp
+	}
+
+	truncatedContactInfo := utils.TruncateString(contactInfo, 3)
+	contactTypeStr := utils.Humanize(string(contactType))
+	log.Ctx(ctx).Infof("sending OTP message to %s %s...", contactTypeStr, truncatedContactInfo)
+	err = h.MessageDispatcher.SendMessage(ctx, msg, organization.MessageChannelPriority)
+	if err != nil {
+		return fmt.Errorf("cannot send OTP message through %s: %w", contactTypeStr, err)
```

Comment:
Just a suggestion, but do we need to also add the `truncatedContactInfo` to the error to make sure which contact caused this ? I know it's logged in the infof above, but it may be not deterministic in case there are multiple requests at the same time.

Reply:
Addressed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/413/commits/29cbf7238ddf600e5f1e76ac9406092ab62b4f9e.